src/System.IO.FileSystem/src/System/IO/FindEnumerable.Windows.cs

```diff
@@ -0,0 +1,207 @@
+                if (_lastEntryFound)
+                    return false;
+
+                RawFindData findData = default;
+                do
+                {
+                    FindNextFile();
+                    if (!_lastEntryFound && _info != null)
+                    {
+                        // If needed, stash any subdirectories to process later
+                        if (_pending != null && (_info->FileAttributes & FileAttributes.Directory) != 0
+                            && !PathHelpers.IsDotOrDotDot(_info->FileName))
+                        {
+                            string subDirectory = PathHelpers.CombineNoChecks(_currentPath, _info->FileName);
+                            _pending.Enqueue((CreateDirectoryHandle(subDirectory), subDirectory));
+                        }
+
+                        findData = new RawFindData(_info, _currentPath, _originalPath, _originalUserPath);
+                    }
+                } while (!_lastEntryFound && !_predicate(ref findData));
+
+                return !_lastEntryFound;
+            }
+
+            private unsafe void FindNextFile()
+            {
+                Interop.NtDll.FILE_FULL_DIR_INFORMATION* info = _info;
+                if (info != null && info->NextEntryOffset != 0)
+                {
+                    // We're already in a buffer and have another entry
+                    _info = (Interop.NtDll.FILE_FULL_DIR_INFORMATION*)((byte*)info + info->NextEntryOffset);
+                    return;
+                }
+
+                // We need more data
+                if (GetData())
+                    _info = (Interop.NtDll.FILE_FULL_DIR_INFORMATION*)_pinnedBuffer.AddrOfPinnedObject();
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public void NoMoreFiles()
+            {
+                _info = null;
+                if (_pending == null || _pending.Count == 0)
+                {
+                    _lastEntryFound = true;
+                }
+                else
+                {
+                    // Grab the next directory to parse
+                    var next = _pending.Dequeue();
+                    Interop.Kernel32.CloseHandle(_directoryHandle);
+                    _directoryHandle = next.Handle;
+                    _currentPath = next.Path;
+                    FindNextFile();
+                }
+            }
+
+            public void Reset()
+            {
+                throw new NotSupportedException();
+            }
+
+            public void Dispose()
+            {
+                Dispose(disposing: true);
+                GC.SuppressFinalize(this);
+            }
+
+            protected void Dispose(bool disposing)
+            {
+                byte[] buffer = Interlocked.Exchange(ref _buffer, null);
+                if (buffer != null)
+                    ArrayPool<byte>.Shared.Return(buffer);
+
+                var queue = Interlocked.Exchange(ref _pending, null);
```

Comment:
Nit: var should be replaced by the actual type

Reply:
Here I can't afaik. It doesn't allow me to use `.Handle` without redefining.

Reply:
Can you elaborate?  What code did you write and what error did you get?  You should be able to just write:
```C#
Queue<(IntPtr Handle, string Path)> queue = Interlocked.Exchange(ref _pending, null);
```
That doesn't work?

Reply:
That does, you just have to redefine the names again. With var you don't.

Reply:
> With var you don't.

But with `var` as I'm reading the code I have no idea what this thing is.